go-controller/pkg/networkmanager/network_controller.go

```diff
@@ -124,84 +227,296 @@ func (c *networkController) getAllNetworkStates() []*networkControllerState {
-			klog.V(5).Infof(
-				"ignoring network %q since %q does not manage it",
-				network.GetNetworkName(),
-				c.name,
-			)
-		} else if err != nil {
-			return fmt.Errorf("failed to sync network %s: %w", network.GetNetworkName(), err)
+	if !have.stoppedAndDeleting {
+		have.controller.Stop()
+	}
+	have.stoppedAndDeleting = true
+
+	err := have.controller.Cleanup()
+	if err != nil {
+		return fmt.Errorf("%s: failed to cleanup network %s: %w", c.name, network, err)
+	}
+
+	c.setNetworkState(network, nil)
+	return nil
+}
+
+func (c *networkController) gatherNetwork(network util.MutableNetInfo) error {
+	if network == nil {
+		return nil
+	}
+	return c.setAdvertisements(network)
+}
+
+func (c *networkController) setAdvertisements(network util.MutableNetInfo) error {
+	if !network.IsDefault() && !network.IsPrimaryNetwork() {
+		return nil
+	}
+	if !c.hasRouteAdvertisements() {
+		return nil
+	}
+
+	raNames := sets.New[string]()
+	for _, nadNamespacedName := range network.GetNADs() {
+		namespace, name, err := cache.SplitMetaNamespaceKey(nadNamespacedName)
+		if err != nil {
+			return err
 		}
+
+		nad, err := c.nadLister.NetworkAttachmentDefinitions(namespace).Get(name)
+		if err != nil {
+			return err
+		}
+
+		var nadRANames []string
+		if nad.Annotations[types.OvnRouteAdvertisementsKey] != "" {
+			err = json.Unmarshal([]byte(nad.Annotations[types.OvnRouteAdvertisementsKey]), &nadRANames)
+			if err != nil {
+				return err
+			}
+		}
+
+		raNames.Insert(nadRANames...)
 	}
-	klog.Infof("%s: finished syncing all networks. Time taken: %s", c.name, time.Since(start))
+
+	podAvertisements := map[string][]string{}
+	eipAdvertisements := map[string][]string{}
+	for raName := range raNames {
+		ra, err := c.raLister.Get(raName)
+		if err != nil {
+			return err
+		}
+
+		if !ra.Spec.Advertisements.PodNetwork {
+			continue
+		}
+
+		accepted := meta.FindStatusCondition(ra.Status.Conditions, "Accepted")
+		if accepted == nil {
+			// if there is no status we can safely ignore
+			continue
+		}
+		if accepted.Status != metav1.ConditionTrue || accepted.ObservedGeneration != ra.Generation {
+			// if the RA is not accepted, we commit to no change, best to
+			// preserve the old config while we can't validate new config
+			return fmt.Errorf("failed to reconcile network %q: RouteAdvertisements %q not in accepted status", network.GetNetworkName(), ra.Name)
+		}
+
+		nodeSelector, err := metav1.LabelSelectorAsSelector(&ra.Spec.NodeSelector)
+		if err != nil {
+			return err
+		}
+
+		nodes, err := c.nodeLister.List(nodeSelector)
+		if err != nil {
+			return err
+		}
+
+		vrf := ra.Spec.TargetVRF
+		if vrf == "" {
+			vrf = types.DefaultNetworkName
+		}
+
+		for _, node := range nodes {
+			if !c.isNodeManaged(node) {
+				continue
+			}
+			if ra.Spec.Advertisements.PodNetwork {
+				podAvertisements[node.Name] = append(podAvertisements[node.Name], vrf)
```

Comment:
i see now adv map containing nodename and tartget vrfs as the values. any reason why it must be vrf list ?